tests/integ/modin/index/test_name.py

```diff
@@ -317,3 +317,39 @@ def test_index_SNOW_1021837():
     native_df_reset = native_df.reset_index()
     snow_df_reset = snow_df.reset_index()
     assert_frame_equal(snow_df_reset, native_df_reset)
+
+
+@sql_count_checker(query_count=0)
+def test_index_non_list_like_names():
+    """
+    Bug SNOW-1650853:
+    Previously, setting names to non-list-like values raised a TypeError: object of type 'int' has no len().
+    That was not a clear error message and therefore has been changed to match native pandas' ValueError:
+    Names must be a list-like.
+    """
+    native_df = native_pd.DataFrame(list(range(10)))
+    snow_df = pd.DataFrame(native_df)
+    eval_snowpark_pandas_result(
```

Comment:
Is there a reason to use eval_snowpark_pandas_result here?  If you use pytest.raises you can also match the expected error message which seems like the objective here.

Reply:
I used `eval_snowpark_pandas_result` because the exact same error is raised and we can compare that behavior to native pandas